artifactory/services/utils/properties.go

```diff
@@ -9,92 +9,142 @@ import (
 	"strings"
 )
 
-type PropertyParseOptions int
-
-const (
-	// Parsing properties
-	SplitCommas PropertyParseOptions = iota
-	JoinCommas
-)
+const PropsSeparator = ";"
+const ValuesSeparator = ","
 
 type Properties struct {
-	Properties []Property
+	properties map[string][]string
 }
 
 type Property struct {
 	Key   string
 	Value string
 }
 
+func NewProperties() *Properties {
+	return &Properties{properties: make(map[string][]string)}
+}
+
 // Parsing properties string to Properties struct.
-func ParseProperties(propStr string, option PropertyParseOptions) (*Properties, error) {
-	props := &Properties{}
-	propList := strings.Split(propStr, ";")
+func ParseProperties(propStr string) (*Properties, error) {
+	props := NewProperties()
+	err := props.ParseAndAddProperties(propStr)
+	return props, err
+}
+
+func (props *Properties) ParseAndAddProperties(propStr string) error {
+	propList := strings.Split(propStr, PropsSeparator)
 	for _, prop := range propList {
 		if prop == "" {
 			continue
 		}
 
-		key, values, err := splitProp(prop)
-		if err != nil {
-			return props, err
+		parts := strings.Split(prop, "=")
+		if len(parts) != 2 {
+			return errorutils.CheckError(errors.New("Invalid property: " + prop))
```

Comment:
"Invalid property: " + prop
-->
fmt.Sprintf("Invalid property format: %s - Format should be key=val1,val2, ...", prop)